packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -391,11 +391,30 @@ export type SdkModelPropertyType =
   | SdkBodyParameter
   | SdkHeaderParameter;
 
+export interface MultipartOptionsType {
+  // whether name is defined in Typespec. For multipart/mixed, name may not be defined for some parts
+  isNameDefined: boolean;
+  // whether this part is for file
+  isFilePart: boolean;
+  // whether this part is multi in request payload
+  multi: boolean;
```

Comment:
just curious why we need this? I mean we already under the interface `MultipartOptionsType`.

Reply:
backward compitibility?

Also, I am not sure if `MultipartOptionsType` exists, if "single" (I assume our existing cadl-ranch cases is such).

Reply:
It is easy to understand with design doc: https://github.com/Azure/typespec-azure/pull/987/

Reply:
- For `addresses: HttpPart<AddressModel[]>`, TCGC API:
```
"name": "addresses",
"type": {"kind": "array"}
"multi": false,
```
It means this part could contain array of model
- For `addresses: HttpPart<AddressModel>[]`, TCGC API:
```
"name": "addresses",
"type": {"kind": "model"}
"multi": true,
```
It means each part could only contain content of one model.

Reply:
Isn't the information redondant with the `kind` in `type`?